lib/graphql_schema.rb

```diff
@@ -24,6 +24,10 @@ def types
     @types ||= @hash.fetch('types').map{ |type_hash| TypeDefinition.new(type_hash) }.sort_by(&:name)
   end
 
+  def type_from_name(type_name)
+    @types.find { |type| type.name == type_name }
```

Comment:
Making this convenient to call a lot could also make a code generator slow.  Would it make sense to build and memoize a hash mapping name to type instead?